### PR TITLE
main.go integration test for Startup interrupting.

### PR DESCRIPTION
**Pull request by krasi-georgiev**
covers issues like: https://github.com/prometheus/prometheus/issues/2563

### CODE REVIEW

**Comment by krasi-georgiev** [comment]
@grobie , @Gouthamve - what do you think about this one?

File: cmd/prometheus/main_test.go [hunk 1]
```diff
@@ -14,11 +14,98 @@
...
+	var err error
+	promPath, err = os.Getwd()
+	if err != nil {
+		fmt.Printf("can't get current dir :%s \n", err)
+		os.Exit(1)
+	}
+	promPath = filepath.Join(promPath, "prometheus")
+
+	flag.Parse()
+	if !testing.Short() {
```
Comment (Gouthamve) [inline cmd/prometheus/main_test.go]: Can we have 
```
if testing.Short() {
  t.Skip()
}
```

Comment (krasi-georgiev) [reply cmd/prometheus/main_test.go]: nice catch about the /data folder , I forgot about this , thanks.

**Comment by krasi-georgiev** [comment]
@Gouthamve PTAL

**Pull request merged**